tests/optimize/test_qng.py

```diff
@@ -73,6 +73,7 @@ def circuit(params):
         assert np.allclose(step1, expected_step)
         assert np.allclose(step2, expected_step)
 
+    @pytest.mark.usefixtures("use_legacy_opmath")  # currently fails with new opmath
     def test_step_and_cost_autograd_with_gen_hamiltonian(self):
```

Comment:
I believe the strategy for this is to have two test cases, and xfail the failing one?

Reply:
Yep I created a story for that https://app.shortcut.com/xanaduai/story/58893/update-tests-tests-optimize-test-qng-py-testoptimize-test-step-and-cost-autograd-with-gen-hamiltonian

Reply:
Oh but yes you are right I didnt at an xfailing test, only for the lightning one